samples/oauth_rails/VetsApiClaims/app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,89 @@
+class DocumentsController < ApplicationController
+  before_action :require_auth
+  before_action :setup_from_session
+
+  def index
+    @documents = Document.all.order(created_at: :asc) # where(username: @session.id_token_attributes[:preferred_username])
+  end
+
+  def new
+  end
+
+  def create
+    res = RestClient.post("https://dev-api.va.gov/services/vba_documents/v0/uploads", nil,  apikey: Figaro.env.va_api_key)
+    parsed = JSON.parse(res.body)['data']
+    Document.create(
+      guid: parsed['attributes']['guid'],
+      status: parsed['attributes']['status'],
+      location: parsed['attributes']['location'],
+      username: @session.id_token_attributes[:preferred_username]
+    )
+    redirect_to documents_path
+  end
+
+  def edit
+    @document = Document.find params[:id]
+  end
+
+  def update
+    @document = Document.find params[:id]
+    file_contents = File.read( params[:upload].tempfile.path )
+    content = build_content(file_contents.chop)
+    encoded = Base64.encode64(content)
+    RestClient.put(@document.location,
+                         "data:multipart/form-data;base64,#{encoded}",
+                         {'Content-Encoding': 'base64',
+                          'Content-Type': 'application/pdf'
+    }
+                    )
+    redirect_to documents_path
+  end
+
+  def update_status
+    document = Document.find params[:id]
+    res = RestClient.get("https://dev-api.va.gov/services/vba_documents/v0/uploads/#{document.guid}",  apikey: Figaro.env.va_api_key)
+    parsed = JSON.parse(res.body)['data']
+    document.update(
+      status: parsed['attributes']['status'],
+      code: parsed['attributes']['code'],
+      detail: parsed['attributes']['detail'],
+      updated_at: parsed['attributes']['updated_at']
+    )
+    redirect_to documents_path
+  end
+
+  def download
+    document = Document.find params[:id]
+    raw = RestClient::Request.execute(
+      method: :get,
+      url: "https://dev-api.va.gov/services/vba_documents/v0/uploads/#{document.guid}/download",
+      raw_response: true,
+      headers: {
+        apikey: Figaro.env.va_api_key
+      }
+    )
```

Comment:
Might make sense to abstract out, but I like being verbose in the sample apps because its a more clear code example. As for v0 vs v1, I never added the permission to my api user :(

Reply:
:P :P I say upgrade to v1